TEST_APPS/testcases/nfc/test_nfc.py

```diff
@@ -0,0 +1,288 @@
+
+    response = self.nfc_command("dev1", "iseeprom")
+    eeprom = response.parsed['iseeprom']
+    self.logger.info("Target includes NFCEEPROM: %s" % eeprom)
+
+    self.nfc_command("dev1", "initnfc")
+    if not eeprom:
+        self.nfc_command("dev1", "start")
+    self.nfc_command("dev1", "erase")
+    self.nfc_command("dev1", "writelong %d %s" % (textLength,messageRep))
+
+    self.clf.parse("connect")
+    tag = self.clf.clf_response()
+    asserts.assertNotNone(tag, "Could not connect to any tag")
+
+    # assert that read the eeprom contents gives correct data and length
+    asserts.assertEqual(tag.ndef.records[0].__class__.__name__, "TextRecord", "expected TextRecord")
+    asserts.assertEqual(len(tag.ndef.records[0].text), len(expected_text))
+    self.assert_text_equal(tag.ndef.records[0].text, expected_text, len(expected_text))
+
+
+@test_case(CreamSconeTests)
+def test_nfce2e_reprogrammed_stress(self):
+    """
+    check - Large record can be programmed from a remote and read via contactless
+    """
+    messageRep = 'thequickbrownfoxjumpedoverthelazydog' # repeating message written
+    textLength = 204       # 2K < x < 4K
+
+    # calculate actual message to compare to using the library
+    message = nfc_messages.make_textrecord( nfc_messages.repeat_string_to_length(messageRep, textLength))
+    expected_message = str(message)
+
+    response = self.nfc_command("dev1", "iseeprom")
+    eeprom = response.parsed['iseeprom']
+    self.logger.info("Target includes NFCEEPROM: %s" % eeprom)
+
+    self.nfc_command("dev1", "initnfc")
+    if not eeprom:
+        self.nfc_command("dev1", "start")
+    self.nfc_command("dev1", "erase")
+
+    # program a large tag to target wirelessly
+    self.clf.parse("connect")
+    tag = self.clf.clf_response()
+    asserts.assertNotNone(tag, "Could not connect to any tag")
+    nfc_messages.program_remote_tag(message, tag)
+    self.logger.info("%d bytes chunk of data written to tag remotely" %  len(str(message)))
+
+    # read device
+    self.clf.parse("connect")
+    tag = self.clf.clf_response()
+    asserts.assertNotNone(tag, "Could not re-connect to any tag")
+    self.clf.parse("mute") # disable the reader radio
+
+    # verify in target
+    response = self.nfc_command("dev1", "readmessage")
+    asserts.assertEqual(len(response.parsed['nfcmessage']), len(expected_message))
+    self.assert_binary_equal(response.parsed['nfcmessage'], expected_message, len(expected_message))
+
+
+@test_case(CreamSconeTests)
+def test_nfce2e_discovery_loop(self):
+    """
+    check - Controller discovery loop stop/start
+    fails : blocked by an issue
+    """
+    expectedURI = "https://www.nasa.com"    # ensure that these differ per test case
+
+    response = self.command("dev1", "iseeprom") # will hold result from the init lib call
+    eeprom = response.parsed['iseeprom']
+    self.logger.info("Target includes NFCEEPROM: %s" % eeprom)
+
+    self.nfc_command("dev1", "initnfc")
+
```

Comment:
Where is the discovery loop started/stopped?

Reply:
ticket raised , stop breaks.